ecommerce/serializers.py

```diff
@@ -202,6 +204,24 @@ class ProductDetailSerializer(ProductSerializer):
 
     latest_version = serializers.SerializerMethodField()
 
+    @staticmethod
+    def setup_eager_loading(queryset):
+        """
+        setup django eager loading.
+        """
+        queryset = queryset.select_related("content_type")
+        courserun_ct = ContentType.objects.get_for_model(CourseRun)
+        tags_with_courserun = [item for item in queryset if item.content_type_id == courserun_ct.id]
+        prefetch_related_objects(tags_with_courserun, "content_object__course__courseruns")
+        
+        queryset = queryset.prefetch_related(
```

Comment:
If removing the `prefetch_related_objects` as commented previously, this could simply go `queryset.select_related(...).prefetch_related(...)`